src/index.js

```diff
@@ -0,0 +1,250 @@
+ *  text: string;
+ *  recipients: string;
+ * }} EmailMessage
+ */
+
+/**
+ * @typedef {{ 
+*  type: 'ifttt';
+*  event: string;
+*  value1: string;
+*  value2: string;
+*  value3: string;
+* }} IftttMessage
+*/
+
+/**
+* @typedef {{ 
+*  type: 'pushcut';
+*  notification: string;
+*  title: string;
+*  text: string;
+*  input: string;
+*  actions: string;
+* }} PushcutMessage
+*/
+
+/**
+ * @typedef {{ 
+ *  type: 'pushover';
+ *  name: string;
+ *  priority: number;
+ *  device: string;
+ *  sound: string;
+ *  url: string;
+ *  urltitle: string;
+ * }} PushoverMessage
+*/
+
+/**
+ * @typedef {EmailMessage | IftttMessage | PushcutMessage | PushoverMessage} HomebridgeMessage
+ */
+
+class HomebridgeMessenger {
+    /**
+     * @param {import("homebridge").Logger} log
+     * @param {import("homebridge").AccessoryConfig} config
+    */
+    constructor (log, config) {
+        this.log = log
+        this.config = config
+        /** @type {HomebridgeMessage[]} */
+        this.messages = this.config.messages ?? []
+        /** @type {import("homebridge").API['hap']['Service']['Switch'][]} */
+        this.serviceMessagesSwitches = []
+
+        // Add main switch to Homebridge
+        this.serviceMainSwitch = new Service.Switch(this.config.name, 0)
+        this.logMessage("Added Main Switch: " + this.config.name)
+
+        // Initialize cache
+        this.cacheDirectory = HomebridgeAPI.user.persistPath()
+        this.storage = require("node-persist")
+    
+        /** @type {Parameters<import("node-persist").LocalStorage['init']>[0]} */
+        const storagePersistOptions = {
+            dir: this.cacheDirectory,
+            forgiveParseErrors: true,
+        }
+
+        this.storage.initSync(storagePersistOptions)
+
+        // Get cache and validate if main switch is in cache
+        const itemInCache = this.storage.getItemSync(this.config.name)
+        this.on = !!itemInCache
+
+        if (!itemInCache) {
+            this.serviceMainSwitch.setCharacteristic(Characteristic.On, false)
+            this.logMessage("Main Switch: OFF")
+        } else {
+            this.serviceMainSwitch.setCharacteristic(Characteristic.On, true)
+            this.logMessage("Main Switch: ON")
+        }
+
+        this.loadConfiguredMessages()
+    }
+
+    /**
+     * @param {string} message
+     */
+    logMessage(message) {
+        this.log(taggedMessage(message))
+    }
+
+    loadConfiguredMessages() {
+        for (const [index, message] of this.messages.entries()) {
+            const {name: messageName, type: _messageType} = message
+
+            /** @type {HomebridgeMessage['type']} */
+            const messageType = _messageType.toLowerCase()
+
+            const serviceMessageSwitch = new Service.Switch(messageName , index + 100)
+            this.logMessage(`Added ${messageType}: ${messageName}`)
+
+            serviceMessageSwitch.getCharacteristic(Characteristic.On)
+                .on("set", function (value, callback) {
```

Comment:
I was tempted to wrap most of the `"set"` handler in a `try-catch`,  but I leaned towards trying to keep the code functioning the same. It's possible for the function to throw an error when a messenger is instantiated and when the message is sent

(Though most of the messengers don't actually propagate the errors that happen in `sendMessage` up to the caller)